### PR TITLE
fix(premissions)_: convert balance from GetCachedBalancesByChain

### PR DESCRIPTION
**Pull request by mprakhov**
Fix converting balance from GetCachedBalancesByChain

Closes # https://github.com/status-im/status-desktop/issues/15150

### CODE REVIEW

File: services/wallet/token/balance_persistence_test.go [hunk 1]
```diff
@@ -161,3 +164,87 @@ func TestSaveTokens(t *testing.T) {
...
+	require.NoError(t, err)
+	require.Len(t, result, 1)
+
+	require.Equal(t, result[chain1][address1][tokenAddress1].ToInt(), big.NewInt(1))
+
+	result, err = tokenManager.GetCachedBalancesByChain([]common.Address{address1, address2}, []common.Address{tokenAddress2, tokenAddress1}, []uint64{chain1, chain2})
+	require.NoError(t, err)
+	require.Len(t, result, 2)
+
+	require.Equal(t, result[chain1][address1][tokenAddress1].ToInt(), big.NewInt(1))
```
Comment (MishkaRogachev) [inline services/wallet/token/balance_persistence_test.go:251]: minor: use value directly from `token2.BalancesPerChain[chain2]` (for all `Equal` calls)

**Review by MishkaRogachev — APPROVED** [review]
Nice!

File: services/wallet/token/balance_persistence_test.go [hunk 2]
```diff
@@ -6,6 +6,9 @@ import (
 
 	"github.com/stretchr/testify/require"
 
+	"github.com/status-im/status-go/rpc/network"
+	"github.com/status-im/status-go/services/wallet/community"
+	"github.com/status-im/status-go/services/wallet/token"
```
Comment (kounkou) [inline services/wallet/token/balance_persistence_test.go]: Do you need to import wallet/token here ? There must be something wrong here. Could you check please

File: services/wallet/token/balance_persistence_test.go [hunk 3]
```diff
@@ -161,3 +164,87 @@ func TestSaveTokens(t *testing.T) {
...
+	require.Len(t, result, 1)
+
+	require.Equal(t, result[chain1][address1][tokenAddress1].ToInt(), big.NewInt(1))
+
+	result, err = tokenManager.GetCachedBalancesByChain([]common.Address{address1, address2}, []common.Address{tokenAddress2, tokenAddress1}, []uint64{chain1, chain2})
+	require.NoError(t, err)
+	require.Len(t, result, 2)
+
+	require.Equal(t, result[chain1][address1][tokenAddress1].ToInt(), big.NewInt(1))
+	require.Equal(t, result[chain2][address2][tokenAddress2].ToInt(), big.NewInt(1000000000000000000))
```
Comment (kounkou) [inline services/wallet/token/balance_persistence_test.go:252]: I wish there is a go-standard way to humanize this. Too many zeroes here... 🥲

**Comment by kounkou** [comment]
After checking the build I can see the failure due to the imports matching my comment. This is my only blocking comment. Please fix.

**Comment by mprakhov** [comment]
> After checking the build I can see the failure due to the imports matching my comment. This is my only blocking comment. Please fix.

There was a linter issue. Before the merge I did a rebase and it broke the test (changes on the wallet side). I've fixed it

**Review by kounkou — APPROVED** [review]
LGTM

File: services/wallet/token/balance_persistence_test.go (hunk 2 above)
Comment (mprakhov) [reply services/wallet/token/balance_persistence_test.go]: yes, I need this, all is ok, in other case linter will not allow me to pass
also, on save and non used imports are removed automatically

File: services/wallet/token/balance_persistence_test.go (hunk 3 above)
Comment (mprakhov) [reply services/wallet/token/balance_persistence_test.go:252]: we convert the result to BigInt during checking permission, that's why I added the test, where I do this conversion, to ensure we get the correct value.

**Pull request merged**